xla/service/gpu/buffer_comparator.cc

```diff
@@ -182,49 +183,56 @@ static absl::StatusOr<bool> CompareEqualParameterized(
 absl::StatusOr<bool> BufferComparator::CompareEqual(
     se::Stream* stream, se::DeviceMemoryBase current,
     se::DeviceMemoryBase expected) const {
+
+  ComparisonParams params{
+    .relative_tol = relative_tol_,
+    .verbose = verbose_,
+    .shape = shape_, // Keep only reference to the shape
+    .stream = stream,
+    .current = current,
+    .expected = expected,
+  };
+
+#define FUNC_NAME(x) #x, buffer_comparator::x()
```

Comment:
Sorry for adding one more comment, this macro which generates 2 values is really not a good fit for our code style, could you just write out the names twice, as it was?
https://google.github.io/styleguide/cppguide.html#Preprocessor_Macros
"Try not to use macros that expand to unbalanced C++ constructs..."